### ISSUE TITLE
mesa-vulkan-drivers bites proton on nvidia platform

### ISSUE THREAD

**Issue by ardje** [ValveSoftware/SteamOS#652]
#### Your system information

* Steam client version: 2018-09-26
* SteamOS version: brewmaster beta
* Opted into Steam client beta?: Yes 
* Opted into SteamOS beta?: Yes 
* Have you checked for updates in Settings > System?: Yes
(zotac sn970 steam machine)

#### Please describe your issue in as much detail as possible:
With a few workarounds I had GTA V working:
https://github.com/ValveSoftware/Proton/issues/37

Since 20th of september GTA V started throwing DX10/11 errors.
This was after the update of a new beta client.
I also had a new steam package and a new steam-launcher package installed, but I doubt they were a problem

Anyway: I boiled the problem down to this: Proton was unable to select the right vulkan driver, although vulkaninfo just gives a lot of warnings and dumps the nvidia vulkan info.
But I took a good look at those warnings: it said it could not use the amd icd, nor could it use the amd and intel vulkan so files.
I noticed that reinstalling the nvidia-vulkan-icd package also installs nothing.
So I opted for the last thing:
apt remove mesa-vulkan-drivers.
So why is this a steamos bug:
steamos-packages is depending on mesa-vulkan-drivers.
https://github.com/ValveSoftware/Proton/issues/37#issuecomment-425213267

Afaik, removing mesa-vulkan-drivers is an easy way out, the only real thing I thinkg that should be done is to have a real nvidia icd, and mv the amd icd's to disabled or whatever.
But currently nvidia+mesa-vulkan-drivers+latest steam client on steamos brewmaster_beta just does not get configured correctly.

**Comment by kisak-valve** [comment]
Looks like the nvidia vulkan icd ends up in [nvidia-vulkan-common](http://repo.steampowered.com/steamos/pool/non-free/n/nvidia-graphics-drivers/nvidia-vulkan-common_396.54~steamos-0+steamos1+bsos1_amd64.deb) and not nvidia-vulkan-icd.

**Comment by ardje** [comment]
Hah, thanks...
I will install mesa-vulkan-drivers and steamos-packages again, and ln -s /usr/share/vulkan/icd.d/nvidia_icd.json to /etc/vulkan/icd.d/
and remove the others. I think then there should be no problems, except that not the right icd's get installed in /etc/vulkan/icd.d/
to be clear: there was one called .dpkg-old ...

**Comment by ardje** [comment]
Ok, thanks!
So removing the other files from /etc/vulkan/icd.d/ and symlinking the nvidia one to it solved it.
The question is: why are they there, and how can only putting the valid icd's in place be automated.
(I am not going into which ICD for which card).
And vulkan should allow to have multiple vendors active, does that make it a bug in DXVK?
Or is the bug that the nvidia icd was the only one missing and was the DXVK now doing it correctly with looking up the correct vendor?
Anyway: it works for me now (tm).

**Comment by dubigrasu** [comment]
I think that was supposed to fixed in 2.149 beta?
It was reported as an issue for 2.145:
https://steamcommunity.com/groups/steamuniverse/discussions/1/2595630410189842232/?ctp=2#c1693785035820936257
and apparently fixed in 2.149 beta:
https://steamcommunity.com/groups/steamuniverse/discussions/1/2595630410189842232/?ctp=2#c1693785669851961119

**Comment by ardje** [comment]
I don't think it is the same, as as far as I know, vulkan was working fine on linux games.
GTA V was also working fine though. So I assume around the 20th of september a vendor neutral correct driver selection was implemented using icd files from /etc/vulkan/icd.d/ . Or it has changed from /usr/share/vulkan/icd.d to /etc/vulkan/icd.d
There was a very old nvidia vulkan icd lingering around (.dpkg-old so a remnant of an upgrade).
The big question is:
2) what happens if you install a fresh steamos (as steamos is not used by the masses yet, just lazy programmers like me) on either of the two platforms. This installation the original installation. I even have not fixed the persistent network rules (which should have used pci bus and not mac address), as the macs from eth0 and eth1 are from their installation image.
1) what is the correct behaviour. Where should what files go.

So low priority because it works for me and it is easily fixable. But it should work for the generic dude that just wants a "pc master race" console to play on.

**Comment by linuxgaming** [comment]
With the latest update, this issue is now present in the "stable" Steam client on SteamOS beta.
`rm /etc/vulkan/icd.d/* && ln -s /usr/share/vulkan/icd.d/nvidia_icd.json to /etc/vulkan/icd.d/` still works as a workaround.

**Comment by brain-anti-freeze** [comment]
Same issue here (Doom 2016 with Vulkan enabled stopped launching) the fix
worked for me too.

**Comment by shaphanpena1** [comment]
I can confirm that this issue is in a new steamos install. I tried to play Overcooked which I downloaded as a whitelisted game. I have not opted into beta. Nvidia Gt 1030 GPU installed. As a workaround I set game launch options to
PROTON_USE_WINED3D11=1 %command%
Game works now. Hopefully issue gets fixed in a Steamos update soon. Dxvk vulkan has to be working. Sym linking, terminal commands are great for people who like to tinker with linux. But, for the end user that just wants their kid to able to turn on their Steamos console and just play a game that doesn't crash on startup, the current state is beta

**Comment by Plagman** [comment]
So it only affects winevulkan and not native Vulkan apps? Might be some interaction with the loader or device selection.

**Comment by shaphanpena1** [comment]
Not sure Plagman. Haven't test with native vulkan game, yet. I have one title that's vulkan based I think. Ballistic overkill.

**Comment by shaphanpena1** [comment]
In response to Linux gaming. How do use the following
rm /etc/vulkan/icd.d/* && ln -s /usr/share/vulkan/icd.d/nvidia_icd.json /etc/vulkan/icd.d/
Is it via terminal command, file config edit or big picture mode launch option? I ask because many people are or will be affected by this bug in Steamos.

**Comment by linuxgaming** [comment]
> In response to Linux gaming. How do use the following
> rm /etc/vulkan/icd.d/* && ln -s /usr/share/vulkan/icd.d/nvidia_icd.json /etc/vulkan/icd.d/

I switched to tty (ctrl+alt+f2) and logged in as *desktop*, then executed these commands with `sudo` (you'll have to set a password first, as I think the account is passwordless by default, so can't use sudo). At first I only symlinked nvidia_icd.json but it didn't do anything, so I actually had to remove the "old" files from /etc/vulkan/icd.d/

**Comment by shaphanpena1** [comment]
Linux gaming, did you change password via the terminal in Linux desktop or in the try session?

**Comment by shaphanpena1** [comment]
Linux gaming, did you change password via the terminal in Linux desktop or
in the try session?

**Comment by jozefkucia** [comment]
Please try to get a log with the "vulkan" debug channel enabled. Add "+vulkan" to the `WINEDEBUG` variable in the `user_settings.py` file, e.g. `"WINEDEBUG": "+timestamp,+pid,+tid,+seh,+debugstr,+module,+vulkan",`. See https://github.com/ValveSoftware/Proton#runtime-config-options for more information about debug logs.

Also, running the game with `VK_LOADER_DEBUG=all` in a broken configuration might show useful info.
